crypto/crypto.go

```diff
@@ -24,15 +34,17 @@ type PubKey interface {
 	Bytes() []byte
 	VerifyBytes(msg []byte, sig []byte) bool
 	Equals(PubKey) bool
-	Type() string
+	TypeIdentifier() string
```

Comment:
What is the `TypeIdentifier` used for and how does it differ from `Type`?

Reply:
I found the word Type for a string wasn't ideal. I really wanted to use Type for an Enum of possible key types. So I renamed Type to TypeIdentifier.

Reply:
That's OK. Typically `Type()` would return an enum, but this works too I suppose.